proving/proving.go

```diff
@@ -51,29 +32,22 @@ func Generate(ctx context.Context, ch shared.Challenge, cfg config.Config, logge
 		return nil, nil, err
 	}
 
-	result, err := postrs.GenerateProof(options.datadir, ch, cfg)
+	result, err := postrs.GenerateProof(options.datadir, ch, cfg, 20, options.powScrypt)
 	if err != nil {
 		return nil, nil, fmt.Errorf("generating proof: %w", err)
 	}
 	logger.Info("proving: generated proof")
-	logger.Debug("Nonce: %v, Indicies: %v, K2PoW: %d, K3PoW: %d", result.Nonce, result.Indicies, result.K2Pow, result.K3Pow)
-	indicies, err := compressIndicies(result.Indicies, options.numUnits, cfg)
-	if err != nil {
-		return nil, nil, fmt.Errorf("compressing proof indicies: %w", err)
-	}
+	logger.Debug("Nonce: %v, Indices: %v, K2PoW: %d, K3PoW: %d", result.Nonce, result.Indices, result.K2Pow, result.K3Pow)
 
-	proof := &shared.Proof{Nonce: result.Nonce, Indices: indicies, K2Pow: result.K2Pow, K3Pow: result.K3Pow}
+	proof := &shared.Proof{Nonce: result.Nonce, Indices: result.Indices, K2Pow: result.K2Pow, K3Pow: result.K3Pow}
 	proofMetadata := &shared.ProofMetadata{
 		NodeId:          options.nodeId,
 		CommitmentAtxId: options.commitmentAtxId,
 		Challenge:       ch,
-		BitsPerLabel:    cfg.BitsPerLabel,
 		LabelsPerUnit:   cfg.LabelsPerUnit,
 		NumUnits:        options.numUnits,
 		K1:              cfg.K1,
 		K2:              cfg.K2,
-		N:               cfg.N,
```

Comment:
does `K3` need to be part of the Metadata?

Reply:
Actually, definitely no - K3 is parameter influencing verification only

Reply:
Depends on how you see it. A proof is valid if all K3 indices pass the difficulty check even if some of the K2 labels do not pass it, so a proof can be valid with one K3 parameter and invalid with a different one.